packages/agent/src/identity-api.ts

```diff
@@ -0,0 +1,224 @@
+import type { RequireOnly } from '@web5/common';
+
+import type { AgentDataStore } from './store-data.js';
+import type { Web5PlatformAgent } from './types/agent.js';
+import type { DidMethodCreateOptions } from './did-api.js';
+import type { AgentKeyManager } from './types/key-manager.js';
+import type { IdentityMetadata, PortableIdentity } from './types/identity.js';
+
+import { BearerIdentity } from './bearer-identity.js';
+import { isPortableDid } from './prototyping/dids/utils.js';
+import { InMemoryIdentityStore } from './store-identity.js';
+
+export interface IdentityApiParams<TKeyManager extends AgentKeyManager> {
+  agent?: Web5PlatformAgent<TKeyManager>;
+
+  store?: AgentDataStore<IdentityMetadata>;
+}
+
+export interface IdentityCreateParams<
+  TKeyManager = AgentKeyManager,
+  TMethod extends keyof DidMethodCreateOptions<TKeyManager> = keyof DidMethodCreateOptions<TKeyManager>
+> {
+  metadata: RequireOnly<IdentityMetadata, 'name'>;
+  didMethod?: TMethod;
+  didOptions?: DidMethodCreateOptions<TKeyManager>[TMethod];
+  tenant?: string;
+  store?: boolean;
+}
+
+export function isPortableIdentity(obj: unknown): obj is PortableIdentity {
+  // Validate that the given value is an object that has the necessary properties of PortableIdentity.
+  return !(!obj || typeof obj !== 'object' || obj === null)
+    && 'did' in obj
+    && 'metadata' in obj
+    && isPortableDid(obj.did);
+}
+
+export class AgentIdentityApi<TKeyManager extends AgentKeyManager = AgentKeyManager> {
```

Comment:
A bit trivial or bikesheddy but I find this new naming a downgrade. Before with some things named `Manager` and other things named `API` or `Method` I was able to understand what large interfaces were meant for external consumption, how they were logically grouped together and which interfaces were more minor and how those were logically grouped together. With the new naming all of the minor interfaces and large interfaces are blending together and more difficult to understand

Reply:
I don't know to what extent current naming is depending on the other SDKs but while not required it would be helpful and intuitive to introduce naming by concerns like "Manager" for large instance based classes, `Service` for cross-cutting APIs and other various labeling by concerns like e.g. `Adapter`, `Wrapper`, `Facade`, `Component`, `Entity`, `Delegate`, `Proxy`, `Singleton`, `Model`, etc. It's probably a broader discussion but it's something I've noticed while reviewing and implementing the API defined here